core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -144,69 +145,56 @@ private fun computePaths(
-    if (possiblePathWithoutErrorNoConstraints != null) {
-        for (currentConstraint in constraints) {
-            val res =
-                Pathfinding(GraphAdapter(infra.blockInfra, infra.rawInfra))
-                    .setTimeout(timeout)
-                    .setEdgeToLength { block: BlockId -> infra.blockInfra.getBlockLength(block) }
-                    .addBlockedRangeOnEdges(currentConstraint)
-                    .setRemainingDistanceEstimator(remainingDistanceEstimators)
-                    .runPathfinding(waypoints)
-            if (res == null) {
-                // This way of handling it is suboptimal, but it should be reworked soon
-                val relaxedPathResponse =
-                    runPathfindingPostProcessing(infra, possiblePathWithoutErrorNoConstraints)
-
-                when (currentConstraint::class.java) {
-                    ElectrificationConstraints::class.java -> {
-                        throw NoPathFoundException(
-                            IncompatibleConstraintsPathResponse(
-                                relaxedPathResponse,
-                                IncompatibleConstraints(
-                                    listOf(
-                                        RangeValue(
-                                            Pathfinding.Range(Offset.zero(), Offset.zero()),
-                                            "elec"
-                                        )
-                                    ),
-                                    listOf(),
-                                    listOf()
-                                )
-                            )
-                        )
-                    }
-                    LoadingGaugeConstraints::class.java -> {
-                        throw NoPathFoundException(
-                            IncompatibleConstraintsPathResponse(
-                                relaxedPathResponse,
-                                IncompatibleConstraints(
-                                    listOf(),
-                                    listOf(Pathfinding.Range(Offset.zero(), Offset.zero())),
-                                    listOf()
-                                )
-                            )
-                        )
+    if (
+        possiblePathWithoutErrorNoConstraints != null &&
+            possiblePathWithoutErrorNoConstraints.ranges.isNotEmpty()
+    ) {
+        val elecRanges = mutableListOf<RangeValue<String>>()
+        val gaugeRanges = mutableListOf<Pathfinding.Range<TravelledPath>>()
+        val signalRanges = mutableListOf<RangeValue<String>>()
+        var travelledPathBlockStart =
+            Offset<TravelledPath>(
+                -possiblePathWithoutErrorNoConstraints.ranges.first().start.distance
+            )
+        for (range in possiblePathWithoutErrorNoConstraints.ranges) {
+            for (currentConstraint in constraints) {
+                for (blockedRange in currentConstraint.apply(range.edge)) {
```

Comment:
I wonder if it wouldn't be better if `PathfindingConstraint` could manage multiple edges
(i.e. `apply(ranges.map { it.edge })`) or something like that.
EDIT: good job, you already did it in the following commit with `ExplainedPathfindingConstraint`.

Reply:
Actually I dropped the `ExplainedPathfindingConstraint` interface in amended https://github.com/OpenRailAssociation/osrd/pull/7953/commits/8ee6553e87735672c30538f445ee6e66d877
It may be discussed, but what we want to display is not always directly linked to the constraint, and info are not always available in the constraint itself.
So I kept it into `buildIncompatibleConstraintsResponse()`.

I resolve this and let you reopen an issue on amended commit if needed.